gno.land/pkg/gnoland/app.go

```diff
@@ -107,9 +110,11 @@ func NewAppWithOptions(cfg *AppOptions) (abci.Application, error) {
 		func(ctx sdk.Context, tx std.Tx, simulate bool) (
 			newCtx sdk.Context, res sdk.Result, abort bool,
 		) {
+			// Add last gas price in the context
+			ctx = ctx.WithValue(auth.GasPriceContextKey{}, gpKpr.LastGasPrice(ctx))
```

Comment:
Maybe a refactor is necessary generally, for this and other key-value pairs being put in the std.Context instance, but it seems like data that is always necessary should be part of the Context struct's definition.  This approach still makes sense for storing key-value pairs for module data that is not required.

Reply:
> this approach still makes sense for storing key-value pairs for module data that is not required.
Good point.  We can address the refactor in  a separate PR later